File: JavaScript/0345_reverse_vowels_of_string/solution.js
function reverse_vowels(string) {
    string = string.split("");
    const vowels = ['a','e','i','o','u'];
    let vowel_list = new Array();
    for (let index = 0; index < string.length; index ++) {
        if (vowels.includes(string[index].toLowerCase())) {
            vowel_list.push(string[index]);
            string[index] = 0;
        }
    }
    let current = vowel_list.length - 1;
    for (let index = 0; index < string.length; index ++) {
        if (string[index] == 0) {
            string[index] = vowel_list[current];
            current --;
        }
    }
    return string.join("");
}

let string = "IceCreAm";
console.log(reverse_vowels(string));

File: Python/0345_reverse_vowels_of_string/solution.py
def reverse_vowels(string):
    string = list(string)
    vowels = ['a','e','i','o','u']
    vowel_list = list()
    for index in range(len(string)):
        if string[index].lower() in vowels:
            vowel_list.append(string[index])
            string[index] = 0
    current = len(vowel_list) - 1
    for index in range(len(string)):
        if string[index] == 0:
            string[index] = vowel_list[current]
            current -= 1
    return "".join(string)

string = "IceCreAm"
print(reverse_vowels(string))